test/cpp/end2end/xds/xds_end2end_test.cc

```diff
@@ -521,6 +521,19 @@ grpc_millis NowFromCycleCounter() {
   return grpc_timespec_to_millis_round_down(gpr_now(GPR_CLOCK_MONOTONIC));
 }
 
+// There can be milliseconds of difference between cycle time and time spec,
+// when the difference is in the order of a few seconds, we need to know that so
+// that our tests can account for the difference and reduce flakes due to the
+// clock difference.
+grpc_millis ClockDifference() {
+  gpr_cycle_counter now = gpr_get_cycle_counter();
+  grpc_millis cycle_time = grpc_cycle_counter_to_millis_round_up(now);
+  grpc_millis time_spec =
+      grpc_timespec_to_millis_round_down(gpr_now(GPR_CLOCK_MONOTONIC));
```

Comment:
Why are we rounding up for cycle time but rounding down for timespec time?  Shouldn't we do the same thing for both?

Reply:
done